### DISCUSSION TITLE
Access to API without authentication

### DISCUSSION THREAD

**Discussion by kieryk123** [directus/directus#2826] [Archive]
Hi, I am building a website with Next.js and Directus CMS. Can I get access to API without login? How can I make the API public? Is it possible?

**Comment by benhaynes** [comment]
Yes, you would set the permissions for the `public` role — which determines what is available without auth.

**Reply by buckhornmktg** [reply]
I'm trying to allow create access on enrollments and read access on providers. I set the permissions on those tables for the public role and I'm still getting "code":3,"message":"Unauthorized. Are there other permissions I need to give to the public role so this will work? Could this be a bug?

**Reply by benhaynes** [reply]
New roles have default system permissions — did you change/delete any of those? Also, you could be trying to fetch relational data that doesn't have the proper permissions. Or it could be a bug... but fear not, these inconsistencies/issues are resolved in the upcoming Directus 9.

**Reply by buckhornmktg** [reply]
I'm working with a clean install to test. I changed some permissions on the public role so that unauthenticated users could submit a front end from (like a survey or enrollment). The only system permissions I would have changed would be to limit users to "mine only". ...Which also doesn't work, the only user that's shown across the site is the current one, not users that that the current user created or owns. 

However in this case, what I'm trying to do is access the API publicly without an authenticated user. I want people to fill out a front end form and I want to save that to the collection. If a user doesn't exist, I'm assuming the unauthenticated user would inherit the permissions from role 2. Is that a correct assumption? Or, does the user still need to be authenticated and assigned the public role to inherit those permissions? I tried both ways to no avail. 

And speaking of those default permissions. Any users with a new default role can access all of the settings in the admin (by typing in the url). They can't save them, but they can view everything. I'm guessing a user needs full read access to the settings, to make some things work, but they should not be able to view these settings in the app or in the api data (I have not checked to see if the api will leak all this info, it could be an app specific bug).

Yes, I have considered building on the alpha version, due to these issues I've been having with 8, but I have a quickly approaching deadline. I probably won't risk it. I'm trying to hack around these problems with ugly, but functional solutions until version 9 is stable.

**Reply by benhaynes** [reply]
It very well may be that these permissions issues in v8 are caused by a bug. However, our small team is all-in on getting v9 production ready in the next few weeks, so I don't think we (internally) will be able to push a legacy fix for this... though we're happy to review pull-requests!

**Reply by buckhornmktg** [reply]
I totally get that.
